### PR TITLE
UnitTest - Test create group and get group api

### PR DESCRIPTION
**Pull request by at-toannguyen1**
#### Related issues (Required)
- Task #223 
- Bug #

#### Description (Required)
- [x] Create unit test create group and get group api

### CODE REVIEW

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information. (2/22)</td>
    </tr>
  </tbody>
</table>

Test case | Classname|
--- | --- |
Given mock response  - When request createGroup - Then return group | vn.asiantech.way.api.HyperTrackApiTest|
Given mock response - When request getGroupInfo - Then return group | vn.asiantech.way.api.HyperTrackApiTest|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/main/java/vn/asiantech/way/data/model/Group.kt [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
+import com.google.gson.annotations.SerializedName
 import java.io.Serializable
 
 /**
  * Copyright © 2017 Asian Tech Co., Ltd.
  * Created by cuongcaov on 09/11/2017
  */
 data class Group(val id: String, var name: String, var token: String,
-                 var ownerId: String, val createAt: String, var modifiedAt: String) : Serializable
+                 var ownerId: String, @SerializedName("created_at") val createAt: String,@SerializedName("modified_at") var modifiedAt: String) : Serializable
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/model/Group.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/34ce7035196f59be1889db08ee43620a97418a60/app/src/main/java/vn/asiantech/way/data/model/Group.kt#L11"></span></td>
    </tr>
  </tbody>
</table>

File: app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        server.start()
+        val baseUrl = server.url("/api/v1/")
+        restClient = RestClient.getClient(baseUrl, HypertrackService::class.java)
+    }
+
+    @After
+    fun cleanUp() {
+        server.shutdown()
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt]: remove this line

File: app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        /* Given */
+        val test = TestObserver<Group>()
+        server.addResponseBody("createGroup.json")
+
+        /* When */
+        restClient.createGroup("nameGroup").subscribe(test)
+
+        /* Then */
+        val request = server.takeRequest()
+        val body=request.body.readUtf8()
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt]: change to ` val body = request.body.readUtf8()`

File: app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        test.assertValue {
+            assertThat(it.id, `is`("f3ead2ae-dc0a-4a7e-85be-74ee51d9d70a"))
+            assertThat(it.name, `is`("Southern group"))
+            assertThat(it.token, `is`("gk_1409a26018fse12"))
+            assertThat(it.createAt, `is`("2016-03-09T05:20:19.742229Z"))
+            assertThat(it.modifiedAt, `is`("2016-03-09T05:20:19.742257Z"))
+            true
+        }
+    }
+}
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt]: add break line

File: app/src/test/resources/responses/createGroup.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "f3ead2ae-dc0a-4a7e-85be-74ee51d9d70a",
+  "name": "Southern group",
+  "parent_group_id": "15ce9cdc-1409-4976-a260-dc0e5b2445ac",
+  "token": "gk_1409a26018fse12",
+  "created_at": "2016-03-09T05:20:19.742229Z",
+  "modified_at": "2016-03-09T05:20:19.742257Z"
+}
```
Comment (at-vinhhuynh) [inline app/src/test/resources/responses/createGroup.json]: add break line

File: app/src/test/resources/responses/groupInfo.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "83892b6f-4d4e-4ead-bcdd-d95a24571506",
+  "name": "Group Cua Cuong",
+  "lookup_id": null,
+  "parent_group_id": null,
+  "token": "gk_f1a6f81c400dc67bacfd547dfd66aa16b08f8191",
+  "created_at": "2017-11-09T04:02:01.769935Z",
+  "modified_at": "2017-11-09T04:02:01.784838Z"
+}
```
Comment (at-vinhhuynh) [inline app/src/test/resources/responses/groupInfo.json]: add break line

File: app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        server.addResponseBody("createGroup.json")
+
+        /* When */
+        restClient.createGroup("nameGroup").subscribe(test)
+
+        /* Then */
+        val request = server.takeRequest()
+        val body = request.body.readUtf8()
+        assertThat(request.method.toUpperCase(), `is`("POST"))
+        assertThat(body, containsString("nameGroup"))
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt]: change this line to `assertThat(body, 'is'("name=nameGroup"))`

File: app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        server.addResponseBody("createGroup.json")
+
+        /* When */
+        restClient.createGroup("nameGroup").subscribe(test)
+
+        /* Then */
+        val request = server.takeRequest()
+        val body = request.body.readUtf8()
+        assertThat(request.method.toUpperCase(), `is`("POST"))
+        assertThat(body, containsString("name=nameGroup"))
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/HyperTrackApiTest.kt]: change this line to `assertThat(body,  is ("name=nameGroup"))

File: app/src/test/java/vn/asiantech/way/api/CreateGroupTest.kt [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Test
+    fun `Given mock response  - When request createGroup - Then return group`() {
+        /* Given */
+        val test = TestObserver<Group>()
+        server.addResponseBody("createGroup.json")
+
+        /* When */
+        restClient.createGroup("nameGroup").subscribe(test)
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/CreateGroupTest.kt]: change this line to `restClient.createGroup("").subscribe(test)`

File: app/src/test/java/vn/asiantech/way/api/CreateGroupTest.kt [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+        /* Then */
+        val request = server.takeRequest()
+        val body = request.body.readUtf8()
+        Assert.assertThat(request.method.toUpperCase(), CoreMatchers.`is`("POST"))
+        Assert.assertThat(body, CoreMatchers.`is`("name=nameGroup"))
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/CreateGroupTest.kt]: remove all `CoreMatchers.`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

**Pull request merged**